Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/InstanceDataHierarchy.cpp

```diff
@@ -546,7 +546,7 @@ namespace AzToolsFramework
 
             for (auto& element : nodeEditData->m_elements)
             {
-                if (element.IsClassElement() && element.m_elementId == AZ::Edit::ClassElements::Group)
+                if (element.m_elementId == AZ::Edit::ClassElements::Group)
```

Comment:
It would be great if we could slightly expand `Code\Framework\Tests\InstanceDataHierarchy.cpp` to include a test to validate this new "class elements with an associated member" behavior - a test that simply reflects groups with and without an associated member bool variable and validating their layout would be ace!

Reply:
I added unit tests for normal groups, toggle groups, and nested groups!